apps/system/test/marionette/text_selection_test.js

```diff
@@ -265,6 +265,33 @@ marionette('Text selection >', function() {
 
         });
     });
+
+    suite('selection carets bug', function() {
+      setup(function() {
+        fakeTextselectionApp.setTestFrame('bug1120358');
+      });
+      test('bug1120358 : Positions of selection carets should be updated ' +
+           'correctly while scrolling (so far on device only)',
+        function() {
+          fakeTextselectionApp.longPressByPosition('BugContent',30,150);
+          client.executeScript('document.getElementById("bug-content").' +
+                               'scrollTop += 20');
+          client.helper.wait(500);
+          var flickPosition = {};
+          flickPosition.caretB = {
+            offset: {
+              x: 20, y: 0
+            }
+          };
+          fakeTextselectionApp.BugContent.selectionHelper.
+          moveCaretByPosition(flickPosition);
+          client.helper.wait(500);
+          assert.ok(fakeTextselectionApp.BugContent.selectionHelper.
+          selectedContent.length > 0,
```

Comment:
I think we can compare the selectedContent before/after scrolling. What do you think?